app/models/course/assessment/question.rb

```diff
@@ -40,4 +40,24 @@ def attempt(submission)
     return actable.attempt(submission) if actable && actable.self_respond_to?(:attempt)
     fail NotImplementedError, 'Questions must implement the #attempt method for submissions.'
   end
+
+  # Returns the latest correct answer in the given submission for the question.
+  #
+  # @param [Course::Assessment::Submission] submission The submission which the answer should
+  #   belong to.
+  # @return [Course::Assessment::Answer] The latest correct answer of the question
+  # @return [nil] If there is not correct answer
+  def latest_correct_answer(submission)
+    return actable.
+      latest_correct_answer(submission) if actable.self_respond_to?(:latest_correct_answer)
+
+    submission.answers.where(question: self, grade: maximum_grade).last
+  end
+
+  # Checks if the given submission has finished the question.
+  #
+  # @return [Boolean] True if the question has been finished in the submission.
+  def finished_by?(submission)
```

Comment:
"finished" can mean both getting the answer correct and wrong. how about `correct_attempt_in?(submission)`?